### PR TITLE
chore(chaosengine): adding imagePullSecrets & resource in engine-concepts

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Labelled this PR & related issue with `documentation` tag

### CODE REVIEW

File: docs/chaosengine-concepts.md [hunk 1]
```diff
@@ -322,6 +322,33 @@ This section describes the fields in the ChaosEngine spec and the possible value
...
+  <th>Range</th>
+<td><i>user-defined</i> (type: []corev1.LocalObjectReference)</td>
+</tr>
+<tr>
+  <th>Default</th>
+  <td><i>n/a</i></td>
+</tr>
+<tr>
+  <th>Notes</th>
+  <td>The <code>.components.runner.imagePullSecrets</code> allows developers to specify the <code>imagePullSecret</code> name for chaos-runner. </td>
```
Comment (divya-mohan0209) [inline docs/chaosengine-concepts.md]: chaos-runner should be ChaosRunner per the documentation standards we're following.

File: docs/chaosengine-concepts.md [hunk 2]
```diff
@@ -540,6 +594,33 @@ This section describes the fields in the ChaosEngine spec and the possible value
...
 </table>
 
+<table>
+<tr>
+  <th>Field</th>
+  <td><code>.spec.experiments[].spec.components.resources</code></td>
+</tr>
+<tr>
+  <th>Description</th>
+  <td>Specify the resource requirements for the chaos experiment pod</td>
```
Comment (divya-mohan0209) [inline docs/chaosengine-concepts.md]: chaos experiment needs to be rewritten as ChaosExperiment per the documentation standard.

File: docs/chaosengine-concepts.md [hunk 3]
```diff
@@ -540,6 +594,33 @@ This section describes the fields in the ChaosEngine spec and the possible value
...
+  <th>Range</th>
+<td><i>user-defined</i> (type: corev1.ResourceRequirements)</td>
+</tr>
+<tr>
+  <th>Default</th>
+  <td><i>n/a</i></td>
+</tr>
+<tr>
+  <th>Notes</th>
+  <td>The <code>experiment[].spec.components.resources</code> contains the resource requirements for the chaos experiment Pod, where we can provide resource requests and limits for the pod.</td>
```
Comment (divya-mohan0209) [inline docs/chaosengine-concepts.md]: chaos experiment needs to be ChaosExperiment per the documentation standard.

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Please ensure you make the changes per recommendation & adhere to conventions while referring to annotations.

**Comment by ksatchit** [comment]
We can also include spec.component.args - it is missing right now

**Review by ksatchit — APPROVED** [review]

**Comment by ksatchit** [comment]
PR has been updated with review comments. LGTM

**Pull request merged**